src/GHC/RTS/Events/Binary.hs

```diff
@@ -177,6 +178,17 @@ standardParsers = [
                        , ..}
  )),
 
+ (FixedSizeParser EVENT_MEM_RETURN (sz_capset + 3*4) (do
+      heapCapset   <- get
+      currentw     <- get :: Get Word32
+      neededw      <- get :: Get Word32
+      returnedw    <- get :: Get Word32
+      return MemReturn{ current = fromIntegral currentw
+                       , needed = fromIntegral neededw
+                       , returned = fromIntegral returnedw
```

Comment:
Why do we want to cast these `Word32`s to `Int`s here?

Reply:
Yes good point.